src/useD2.js

```diff
@@ -1,27 +1,60 @@
 import { useState, useEffect } from 'react'
-import { init as initD2 } from 'd2'
+import { init as initD2, config as d2Config, getUserSettings } from 'd2'
 import { useConfig } from '@dhis2/app-runtime'
 
 let theD2 = null
 
-export const useD2 = ({ d2Config = {}, onInitialized = () => {} }) => {
-    const { baseUrl, apiVersion } = useConfig()
-    const [d2, setD2] = useState(theD2)
-    const [d2Error, setError] = useState(undefined)
-
-    useEffect(() => {
-        if (!theD2) {
-            initD2({
-                appUrl: baseUrl,
-                baseUrl: `${baseUrl}/api/${apiVersion}`,
-                ...d2Config
-            }).then(async d2 => {
-                await onInitialized(d2)
-                theD2 = d2
-                setD2(d2)
-            }).catch(setError)
-        }
-    }, [])
-
-    return { d2, d2Error } // d2 is null while loading
+const configI18n = async (baseUrl, i18nRoot) => {
+  d2Config.baseUrl = baseUrl
+
+  const settings = await getUserSettings()
+
+  if (settings.keyUiLocale && settings.keyUiLocale !== 'en') {
+    await d2Config.i18n.sources.add(
+      `${i18nRoot}/i18n_module_${settings.keyUiLocale}.properties`
+    )
+  }
+
+  await d2Config.i18n.sources.add('${i18nRoot}/i18n_module_en.properties')
```

Comment:
Technically it would probably be cleaner to use `engine.query` to get the user locale (since it has already been fetched by the shell it will eventually be cached).  Not the end of the world if we want to keep this legacy bit in here, but I've always found the `d2Config` singleton to be a pain.